### ISSUE TITLE
hackage rejects cabal 2.2-supported open source licenses

### ISSUE THREAD

**Issue by l29ah** [haskell/hackage-server#880]
A package containing
```
cabal-version:       2.2
license:             WTFPL
license-file:        LICENSE
```
got kicked out with
```
This server does not accept packages with 'license' field set to e.g.
AllRightsReserved. See https://hackage.haskell.org/upload for more information
about accepted licenses.
```
even though `cabal check` reports it's correct.

**Comment by gbaz** [comment]
@phadej since wtfpl is a valid spdx license, shouldn't we be accepting it? https://www.haskell.org/cabal/release/cabal-latest/doc/API/Cabal/Distribution-SPDX-LicenseId.html#t:LicenseId

**Comment by phadej** [comment]
This is duplicate of https://github.com/haskell/hackage-server/issues/745

**Comment by phadej** [comment]
To clarify, the check is in `hackage-server` code base. Not somewhere in `Cabal` e.g.

**Comment by gbaz** [comment]
Thanks oleg. I think I'll close this as a dup. We should probably change policy to explicitly say OSI approved licenses going forward, to keep things simple.

**Comment by gbaz** [comment]
actually, nice, with https://github.com/haskell/cabal/pull/6878 we can then move to osi or fsf/libre and that'll cover a wider variety of cases.

**Comment by phadej** [comment]
(Note that the patch is will only be released with Cabal-3.4 at the end of Summer, + `hackage-server` migration to use Cabal-3.4. If someone feels it needs to be backported, be my guest, make a PR).
